addons/payment/models/payment_transaction.py

```diff
@@ -631,73 +631,93 @@ def _process_notification_data(self, notification_data):
         """
         self.ensure_one()
 
-    def _set_pending(self, state_message=None):
+    def _set_pending(self, state_message=None, extra_allowed_states=None):
```

Comment:
Following the comment above about tuples in place of lists:
```suggestion
    def _set_pending(self, state_message=None, extra_allowed_states=()):
```
-> which is ok because tuples are immutable.